### DISCUSSION TITLE
Brew does not appear to honor xcode-select configuration during upgrade

### DISCUSSION THREAD

**Discussion by christopherdwelton** [Homebrew/discussions#4122] [Everyday usage] [answered]
### Output of `brew config`

```shell
brew config
HOMEBREW_VERSION: 3.6.19-12-g260ee0e
ORIGIN: https://github.com/Homebrew/brew
HEAD: 260ee0ee6af21d31e8c57f6855d661ff6b325fb9
Last commit: 24 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: fafed6df54fb5211ebee8bb4b2722e4e68295e3c
Core tap last commit: 22 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode-4.2.app/Contents/Developer/usr/bin/git
Curl: 7.85.0 => /usr/bin/curl
macOS: 13.1-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2 => /Applications/Xcode-14.2.app/Contents/Developer
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /usr/local/share/zsh/site-functions/_brew_cask
```

### Description of issue

We have a special Xcode app that we run command line tools from in order to maintain precise version alignment across our mac buildservers.

We select the command line tools using `xcode-select /Applications/Xcode-14.2.app/Contents/Developer`

Brew config correctly picks this up. Despite picking that up, we can not run `brew upgrade` until updating the default command line tools location of `/Library/Developer/CommandLineTools`, with the following error:

```
Error: Your Command Line Tools are too outdated.
Update them from Software Update in System Preferences.
```

Is there anything in `/Library/Developer/CommandLineTools` even being used when the config shows `Xcode: 14.2 => /Applications/Xcode-14.2.app/Contents/Developer` and the version is confirmed by `xcodebuild -version`?

**Comment by SMillerDev** [comment, marked as answer]
Yes, the XCode included tools are not the same as the CommandLine Tools that can be installed in the system and brew builds it's components against the system one to avoid every user needing XCode.

**Reply by christopherdwelton** [reply]
There's no way to ask it to build against the one from it's own config?

**Reply by SMillerDev** [reply]
Well, software rarely builds on your machine. You get pre-built binaries from CI, which reference the CLT. Your current configuration might work just fine until it doesn't. You're way easier off just installing and updating the commandline tools.

**Reply by christopherdwelton** [reply]
Fair enough, thanks for the explanation
